test/lib/return-cycle-dates.lib.test.js

```diff
@@ -0,0 +1,562 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, before, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+const { returnCycleDates } = require('../../app/lib/static-lookups.lib.js')
+
+// Thing under test
+const ReturnCycleDatesLib = require('../../app/lib/return-cycle-dates.lib.js')
+
+describe('Return Cycle Dates lib', () => {
+  const today = new Date()
+  const year = today.getFullYear()
+  const month = today.getMonth()
```

Comment:
It looks like we have to amend our assertions based on what the current date is.

I'd rather make this explicit in the tests that we have to do this by

- controlling what 'today' is in the tests
- having an explicit `describe()`, for example, `"when 'todays' month is greater than the return cycle end date's month"`

Take a look at `let clock = Sinon.useFakeTimers(testDate)` (we use it in some of our tests, for example, `test/lib/general.lib.test.js`) to understand how we can control the date and time within a test.